### ISSUE TITLE
Training loss abnormal

### ISSUE THREAD

**Issue by alan525** [yl-1993/hfsoftmax#6]
Hello, I am using a non-sampling training model, and the loss always changes within a small range[-0.5,0.5]. When using the hf_sample method, the network loss rapidly increases to nan. Does the sample method require a pre-trained model?

**Comment by yl-1993** [comment]
Hi @alan525, the sampling method does not need pre-trained model. Does the non-sampling method work normally in your case? You can try to use small learning for sampling method since the number of sampled classes will influence the scale of gradient. (But in our experiments, it works well with the same learning rate. So I am not very sure the exact reason given current information.
